### PR TITLE
updating GATK

### PR DESCRIPTION
**Pull request by asp8200**
Updating GATK to v4.5.

The test `joint_germline` fails with the new NF 24.1.0-edge. I don't think the failure is due to the upgrade of GATK, as we have seen it on the `dev`-branch before.

This PR also updates samtools in the GATK-markduplicates to v1.19.2. (That was a task left from #1412.) So after this PR has been merged, dev-Sarek will use samtools v1.19.2 everywhere.

### CODE REVIEW

File: subworkflows/local/bam_variant_calling_haplotypecaller/main.nf [hunk 1]
```diff
@@ -31,7 +31,13 @@ workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
...
         .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ interval_name:intervals.baseName, num_intervals:num_intervals, variantcaller:'haplotypecaller' ], cram, crai, intervals, [] ] }
 
-    GATK4_HAPLOTYPECALLER(cram_intervals, fasta, fasta_fai, dict.map{ meta, dict -> [ dict ] }, dbsnp, dbsnp_tbi)
+    GATK4_HAPLOTYPECALLER(
+        cram_intervals,
+        fasta.map{ it -> [[:], it] },
+        fasta_fai.map{ it -> [[:], it] },
+        dict,
+        dbsnp.map{ it -> [[:], it] },
+        dbsnp_tbi.map{ it -> [[:], it] })
```
Comment (maxulysse) [inline subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:40]: can you modify the channels in the upstream subworkflow?

Comment (asp8200) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:40]: Yeah, I was expecting some comment on that 😄 Let me see if I can fix it in the subworkflow.

Comment (asp8200) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:40]: @maxulysse : You want it around [here](https://github.com/nf-core/sarek/blob/2b3517a2567fb9443682ffd498844c52ab66ed33/subworkflows/local/bam_variant_calling_germline_all/main.nf#L129-L135)?

Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:39]: exactly, look at how we do it for CNVKIT for example

**Comment by maxulysse** [comment]
Looking amazing, just one comment left.

Regarding the joint germline failure, I can reproduce the pytest failure locally, but by running the code itself outside of pytest, it does work, so I'd say let's not waste time on this.

**Comment by matthdsm** [comment]
> Looking amazing, just one comment left.
> 
> Regarding the joint germline failure, I can reproduce the pytest failure locally, but by running the code itself outside of pytest, it does work, so I'd say let's not waste time on this.

My experience is migrating from pytest > nf-test fixes a lot of issues with the testing. Perhaps we ca make it pass that way?

**Comment by asp8200** [comment]
Can we merge this as-is and make a gitHub-issue for constructing an nf-test for joint-germline?

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**